PWGHF/TableProducer/candidateCreatorDstar.cxx

```diff
@@ -13,65 +13,413 @@
+#include "PWGHF/Utils/utilsBfieldCCDB.h"
+
+#include "Framework/AnalysisDataModel.h"
 
 using namespace o2;
 using namespace o2::framework;
+using namespace o2::aod::hf_cand_dstar;
+
+namespace o2::aod
+{
+//   using HfDstarsWStatus= soa::Join<aod::HfDstars, aod::HfCutStatusDstar>;
+//   using HfDstarsWStatus_nd_PvRefitInfo =soa::Join<aod::HfDstars, aod::HfCutStatusDstar, aod::HfPvRefitDstar>;
+//   using Hf2ProngsWStatus = soa:: Join<aod::Hf2Prongs, aod::HfCutStatus2Prong>;
+
+using HfDstarsWOStatus = aod::HfDstars;
+using HfDstarsWOStatusAndPvRefitInfo = soa::Join<aod::HfDstars, aod::HfPvRefitDstar>;
+using Hf2ProngsWOStatus = aod::Hf2Prongs;
+} // namespace o2::aod
 
 /// Reconstruction of D* decay candidates
 struct HfCandidateCreatorDstar {
+  Produces<aod::HfD0FromDstarBase> D0CandTable;
+  Produces<aod::HfCandDStarBase> DStarCandTable;
   Configurable<bool> fillHistograms{"fillHistograms", true, "fill histograms"};
 
-  OutputObj<TH1F> hMass{TH1F("hMass", "D* candidates;inv. mass (#pi D^{0}) (GeV/#it{c}^{2});entries", 500, 0., 5.)};
+  // magnetic field setting from CCDB
+  Configurable<std::string> ccdbUrl{"ccdbUrl", "http://alice-ccdb.cern.ch", "url of the ccdb repository"};
+  Configurable<std::string> ccdbPathLut{"ccdbPathLut", "GLO/Param/MatLUT", "Path for LUT parametrization"};
+  Configurable<bool> isRun2{"isRun2", false, "enable Run 2 or Run 3 GRP objects for magnetic field"};
+  Configurable<std::string> ccdbPathGrp{"ccdbPathGrp", "GLO/GRP/GRP", "Path of the grp file (Run 2)"};
+  Configurable<std::string> ccdbPathGrpMag{"ccdbPathGrpMag", "GLO/Config/GRPMagField", "CCDB path of the GRPMagField object (Run 3)"};
+
+  // vertexing
+  // Configurable<bool> DoRefit{"DoPvRefit", false, "Do Primary Vertex Refit. Optional"};
+  Configurable<bool> propagateToPCA{"propagateToPCA", true, "create tracks version propagated to PCA"};
+  Configurable<double> maxR{"maxR", 200., "reject PCA's above this radius"};                                   // ........... what is unit of this?
+  Configurable<double> maxDZIni{"maxDZIni", 4., "reject (if>0) PCA candidate if tracks DZ exceeds threshold"}; // ..........What it DZ?
+  Configurable<double> minParamChange{"minParamChange", 1.e-3, "stop iterations if largest change of any X is smaller than this"};
+  Configurable<double> minRelChi2Change{"minRelChi2Change", 0.9, "stop iterations is chi2/chi2old > this"};
+  Configurable<bool> useAbsDCA{"useAbsDCA", false, "Minimise abs. distance rather than chi2"};
+  Configurable<bool> useWeightedFinalPCA{"useWeightedFinalPCA", false, "Recalculate vertex position using track covariances, effective only if useAbsDCA is true"};
+
+  Service<o2::ccdb::BasicCCDBManager> ccdb; // From utilsBfieldCCDB.h
+  o2::base::MatLayerCylSet* lut;            // From DCAFitterN.h
+  o2::base::Propagator::MatCorrType matCorr = o2::base::Propagator::MatCorrType::USEMatCorrLUT;
+  int runNumber;
+  double bz;
+  float toMicrometers = 10000.; // from cm to µm
```

Comment:
```suggestion
  constexpr float cmToMicrometers = 10000.; // from cm to µm
```

Reply:
error: non-static data member 'cmToMicrometerts' declared 'constexpr'

Reply:
A variable declared as constexpr typically is only used as compile-time constant (but may be stored in read-only memory). Therefore it uses the same convention as enums and uses uppercase CamelCase.

Reply:
How about `static constexpr float CmToMicrometers = 10000.` then?

Reply:
It May work! but why it should be an issue. It is also present in another candidate creator.

Reply:
Because it is a constant and it should be declared as such.